clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -1680,13 +1808,15 @@ private void checkConsumptionStateWhenStart(
      * task state transition in Controller.
      */
     try {
-      // Compare the offset lag is acceptable or not, if acceptable, report completed directly, otherwise rely on the
-      // normal
-      // ready-to-server checker.
+      // For leaders: if offset lag is acceptable, report completed directly, otherwise rely on the
+      // normal ready-to-serve checker. Followers call read-to-serve checker as it looks for leader
+      // completion status as well along with lag.
       boolean isCompletedReport = false;
       long offsetLagDeltaRelaxFactor = serverConfig.getOffsetLagDeltaRelaxFactorForFastOnlineTransitionInRestart();
       long previousOffsetLag = newPartitionConsumptionState.getOffsetRecord().getOffsetLag();
-      if (hybridStoreConfig.isPresent() && newPartitionConsumptionState.isEndOfPushReceived()) {
+      if (hybridStoreConfig.isPresent()
+          && newPartitionConsumptionState.getLeaderFollowerState() == LeaderFollowerStateType.LEADER
```

Comment:
This code is run for OFFLINE to STANDBY transition, leader replicas will not have a chance to run this code IMO. Why adding this check?

Reply:
Removed this change.